drivers/mcp2515/mcp2515.c

```diff
@@ -99,7 +99,9 @@ int mcp2515_init(candev_mcp2515_t *dev, void (*irq_handler_cb)(void *))
         DEBUG("Error setting interrupt pin!\n");
         return -1;
     }
-    gpio_init(dev->conf->rst_pin, GPIO_OUT);
+    if (gpio_is_valid(dev->conf->rst_pin)) {
```

Comment:
How is the reset handled if there is no reset pin set? 
IIRC resetting was quite crucial for this device, and reset problems have caused headaches for more than 1 person.

Reply:
> How is the reset handled if there is no reset pin set? IIRC resetting was quite crucial for this device, and reset problems have caused headaches for more than 1 person.

Personally I'm using two of these [breakouts](https://www.banggood.com/fr/MCP2515-CAN-Bus-Module-Board-TJA1050-Receiver-SPI-51-MCU-ARM-Controller-5V-DC-p-1481199.html?utm_source=googleshopping&utm_medium=cpc_organic&gmcCountry=FR&utm_content=minha&utm_campaign=minha-fr-fr-pc&currency=EUR&cur_warehouse=CN&createTmp=1&utm_source=googleshopping&utm_medium=cpc_eu&utm_content=sandra&utm_campaign=sandra-aceng-ssc-fr-css-all-0423-19bf-v2&ad_id=344815794167&gclid=Cj0KCQjwmPSSBhCNARIsAH3cYgYJR-rUBdbxRQM4gKEdoUwK8uPEVZ3Ab3orJhAwuZbhjuCibmNCKPoaAgzHEALw_wcB) that do not have a reset pin, and they seem to work fine without the reset pin... I guess the initialization procedure works without need for a reset?

Reply:
I'd say that since that breakout board is very widely used It's a good reference to use for this driver. If this one works, green light from me :)

Reply:
I can add the SPI reset call as backup, the datasheet seems to suggest that.